rr_computer_vision/src/traffic_light.cpp

```diff
@@ -1,19 +1,156 @@
-/*
- * @file rr_traffic_light
- * @author Adrian Malaran(AdrianMalaran)
- * @author Toni Ogunmade(oluwatoni)
- * @author Jamie Kim
- * @author Jason Leung
- * @competition IARRC 2018
- */
+  //  @file traffic light detection
+  //  @author Yuchi(Allan) Zhao
+  //  @competition IARRC 2019
 
 #include "traffic_light.hpp"
 
-/**@name TrafficLightDetector
- * @brief Constructor
- */
-TrafficLight::TrafficLight() {
+TrafficLightProcessor::TrafficLightProcessor(ros::NodeHandle nh) : it_(nh_)  {
+    red_light_detected = false;
+    cv::Rect boundRect;
+    red_Pixel_Counter=0;
+    default_ratio=0.0;
+    x=0;
+    y=0;
+    s=0;
+    r=0;
+    redLightCounter=0;
+    greenLightCounter=0;
+    // test_subscriber = it_.subscribe("/zed/left/image_rect_color", 1, &TrafficLightProcessor::TrafficLightImageCallback, this);
+    // test_publisher = it_.advertise("/test_traffic_light", 1);
+    client_ = nh_.serviceClient<std_srvs::Empty>("/Supervisor/start_race");
 }
 
-TrafficLight::~TrafficLight() {
+void TrafficLightProcessor::TrafficLightImageCallback(const sensor_msgs::ImageConstPtr& msg) {
```

Comment:
This callback function is way to big. I think you should chunk up the various steps in this algorithm into different steps and put them into member functions to make it more readable. Otherwise, this looks very daunting and intimidating when you and I both know it is a simple algorithm. Gotta try and organize the code so that people with little context can digest it easily! 

Also Andrew mentioned to me the other day that because of the way callback functions work in ROS you actually can't debug them with breakpoints, which is another strong reason to keep this as small as possible and move as much logic as possible into digestible member functions!